_posts/2020-01-01-Resurrecting-Sigmoid.markdown

```diff
@@ -0,0 +1,521 @@
+
+**Objectives**:
+After doing these readings, we would like you to understand the following background:
+- Explain the vanishing/exploding gradient problem, and why it worsens as networks become deeper
+- Relate vanishing/exploding gradients to the eigenvalues of the input-output Jacobian of the network, and the input-output Jacobian of individual Jacobian layers
+- Explain heuristics used by the community to circumvent the problem of vanishing/exploding gradients, in particular:
+  - common initialization schemes, such as Xavier initialization
+  - pre-training
+  - skip connections / residual neural networks
+  - non-saturating activation functions (ReLU and its variants)
+We would also like you to have an overview of our paper's structure:
+- Motivate and explain the problem the paper is trying to solve: concentrating the entire spectrum of the network's Jacobian around unity
+- Unerstand the setup of the paper, specifically why the following tools are necessary:
+     - Mean-field signal propagation analysis
+     - Random matrix theory
+
+**Topics**:
+
+- Trainability of networks, specifically the vanishing/exploding gradient problem
+- Introduction to the paper / course overview
+
+**Required Reading** 
+
+Prerequisite reading: 
+For those of you that aren’t familiar with deep learning, please read the following sections from the [Deep Learning book](http://www.deeplearningbook.org):
+- 2.7 (Eigendecomposition)
+- 2.8 (Singular value decomposition)
+- 3.2 (Random variables)
+- 3.3 (Probability distributions)
+- 3.7 (Independence and conditional independence)
+- 3.8 (Expectation, variance, and covariance)
+- 5.7 (Supervised learning algorithms)
+
+Initialization: 
+From the [Deep Learning book](http://www.deeplearningbook.org)
+- 8.2 (Challenges in neural network optimization)
+- 8.4 (Parameter initialization strategies)
+
+[All you need is a good init](https://arxiv.org/pdf/1511.06422.pdf) by Mishkin et al., sections 1 and 2
+[Understanding the difficulty of training deep feedforward neural networks](http://proceedings.mlr.press/v9/glorot10a/glorot10a.pdf) by Glorot and Bengio 
+Wikipedia [article](https://en.wikipedia.org/wiki/Residual_neural_network) on residual networks (skip connections)
+
+
+**Optional Reading**:
+
+1. [Deep Residual Learning for Image Recognition](https://arxiv.org/pdf/1512.03385.pdf)
+2. [Depth-first learning](http://www.depthfirstlearning.com/2019/NeuralODEs#3-resnets), section 3 on ResNets
+
+<br />
+
+# 2 Signal propagation
+
+**Motivation**: One of the foundations the analysis in the Resurrecting the Sigmoid paper rests on is the analysis of signal propagation in wide neural networks.  Understanding this mean-field analysis framework, and its results, also connects to more recent investigations of, e.g., neural networks as Gaussian processes, and the neural tangent kernel.  
```

Comment:
nit: Italicize "Resurrecting the Sigmoid"?

Reply:
For "neural networks as Gaussian processes" and "neural tangent kernel", should we hyperlink these to the actual papers?